Realm/Realm/Schema/PropertyTypeEx.cs

```diff
@@ -174,13 +174,47 @@ public static RealmValueType ToRealmValueType(this PropertyType type)
 
         public static bool IsNullable(this PropertyType propertyType) => propertyType.HasFlag(PropertyType.Nullable);
 
-        public static bool IsArray(this PropertyType propertyType) => propertyType.HasFlag(PropertyType.Array);
+        public static bool IsArray(this PropertyType propertyType) => propertyType.HasFlag(PropertyType.Array) && !propertyType.HasFlag(PropertyType.LinkingObjects);
 
         public static bool IsSet(this PropertyType propertyType) => propertyType.HasFlag(PropertyType.Set);
 
         public static bool IsDictionary(this PropertyType propertyType) => propertyType.HasFlag(PropertyType.Dictionary);
 
-        public static bool IsCollection(this PropertyType propertyType) => propertyType.IsArray() || propertyType.IsSet() || propertyType.IsDictionary();
+        public static bool IsCollection(this PropertyType propertyType, out PropertyType collection)
+        {
+            if (propertyType.IsArray())
+            {
+                collection = PropertyType.Array;
+            }
+            else if (propertyType.IsSet())
+            {
+                collection = PropertyType.Set;
+            }
+            else if (propertyType.IsDictionary())
+            {
+                collection = PropertyType.Dictionary;
+            }
+            else
+            {
+                collection = default;
+            }
+
+            return collection != default;
```

Comment:
Would it make sense to name `collection` differently because it can be something that is not a collection?
Or alternatively remove the last `else` and therefore only set it to something that is a `collection` and in the end check for `null`?

Reply:
We are forced to set `collection` to something - the compiler will slap us otherwise. This approach is more or less consistent with other `TryGetXYZ` methods - if the method returns `true`, the `out` parameter will contain a value that is meaningful. If it's `false`, then the out parameter should not be used. So roughly:

```csharp
PropertyType myProp = ...;
if (myProp.IsCollection(out var collection))
{
    // myProp represents some sort of collection
    // the `collection` variable will be set to either Array/Dict/Set
}

// myProp is not a collection
// the `collection` variable is set to the default value
```

Reply:
Understood. Might be just different conventions then. 👍 
Would something like `tryGetAsCollection` be an alternative?

Reply:
I considered that but I wanted it to be consistent with the rest of the `IsXXX` extension methods.

Reply:
Alright. 👍